scripts/g.extension/g.extension.py

```diff
@@ -2319,20 +2321,16 @@ def resolve_source_code(url=None, name=None, branch=None, fork=False):
         module_class = get_module_class_name(name)
 
         # note: 'trunk' is required to make URL usable for 'svn export' call
-        if branch in ["master", "main"]:
-            svn_reference = "trunk"
+        if branch is None or branch in ["master", "main"]:
+            svn_reference = "branches/grass{}".format(version)
```

Comment:
This is confusing. None (default) may be expected to resolve to the default branch which is represented by `trunk` in Subversion interface, although perhaps default resulting in the version branch makes sense in this context. However, explicit master or main should resolve to branch/master and branch/main unless there is a clear reason to do it otherwise.

I though -o is somewhat "behave as with the official repository" which is broader than its defintion "url refers to a fork of the official extension repository". Also depends on what people think fork is. Overall, this might be limiting when you let's say create an institutional repo which should work as the official repository, but you support only one version.